topo/topo.go

```diff
@@ -187,3 +194,37 @@ func GetCollStats(ctx context.Context, m *mongo.Client, db, coll string) (*CollS
 
 	return stats, nil
 }
+
+// RunWithRetry executes the provided function with retry logic for transient errors.
+// It retries the function up to maxRetries times,
+// with an exponential backoff starting from retryInterval.
+func RunWithRetry(
+	ctx context.Context,
+	fn func(context.Context) error,
+	retryInterval time.Duration,
+	maxRetries int,
+) error {
+	if retryInterval <= 0 || maxRetries <= 0 {
+		return errors.New("retryInterval and maxRetries must be greater than zero")
+	}
+
+	var err error
+
+	currentInterval := retryInterval
+	for range maxRetries {
+		err = fn(ctx)
+		if err == nil {
+			return nil
+		}
+
+		if !IsTransientError(err) {
+			return err //nolint:wrapcheck
+		}
+
+		log.Ctx(ctx).Warnf("Transient write error: %v, retrying in %s", err, retryInterval)
+		time.Sleep(currentInterval)
+		currentInterval *= 2
```

Comment:
Not sure about this factor: it'll be 1, 2, 4 sec, which is 7 sec in total. Sometimes it takes 10-20 seconds to elect a primary. So in that case, it would fail. I am not sure if the driver will introduce additional waiting time/timeout, in that case, it might work, but I would test it.

Reply:
Hey, I also thought that this can be too small. Wdyt we should have here, I'm not sure tbh? @sandraromanchenko maybe you can chime in?

Reply:
I see your point and I agree. But I think the factor is OK, essentially a exponential backoff, but `DefaultRetryInterval` of 1 second is probably too small.
I'll set it to 5, that way we will have by default 5, 10, 20..